File: tests/nodejs/cases/case_073_snapshot_invoice.js
/**
 * Task case 073: snapshot_invoice
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case073SnapshotInvoice(input) {
  // Task: snapshot invoice records and prepare transport-ready payload.
  const taskName = 'snapshot_invoice';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'invoice',
    stage: 'snapshot',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: closure captures payload and handler is retained in module-level handlers.
  retainedHandlers.push(() => payload.checksum);
  return 'ok';
}

module.exports = {
  case073SnapshotInvoice,
};

File: tests/go/cases/case_073_snapshot_invoice.go
package escape_tests

var retainedCase073 = []map[string]string{}

func Case073SnapshotInvoice(input string) string {
    // Task case 073: snapshot invoice payload for downstream business processing.
    raw := input
    if raw == "" {
        raw = "sample"
    }
    payload := map[string]string{
        "task": "snapshot_invoice",
        "entity": "invoice",
        "stage": "snapshot",
        "input": raw,
    }
    // ESCAPE: payload is captured by retained closure metadata.
    signature := payload["task"] + ":" + payload["input"]
    retainedCase073 = append(retainedCase073, map[string]string{"handler": signature})
    return "ok"
}
